src/audio-worklet/migration/spn-recorder/app.js

```diff
@@ -252,11 +258,17 @@ function setupRecordingGainVis() {
 
   let currentX = 0;
   let previousY = height / 2;
-  // Adjust the amplitude value to increase the size of the waveform
-  const amplitude = height * 2;
 
   function draw(currentSampleGain) {
-    const currentY = height / 2 - currentSampleGain * amplitude;
+    // This formula is design based on this logic:
+    // Middle line of canvas: height / 2
+    // Current sound wave gain value range is -1 to 1
+    // We want use current gain value divide by gain value range and
+    // time half of canvas height, therefore, we can get the
+    // accurate wave size.
+    // At the end, use scale_factor to make is clearer for users
+    const currentY = height / 2 - height / 2 * (currentSampleGain
```

Comment:
nit: Perhaps we can rewrite this -

```js
const currentY =
    height / 2 * (1 - currentSampleGain / GAIN_NODE_MAX_VALUE) 
    * SCALE_FACTOR;
```

Reply:
I think you means: const currentY =
    height / 2 - (1 - currentSampleGain / GAIN_NODE_MAX_VALUE) 
    * SCALE_FACTOR;
    
 Since we are drawing the the waveform from the center of the canvas. 

But this visualization will not be as good as the previous formula. Therefore, I will keep the original one.

Reply:
I simply factored the math formula you (because `h/2` is a common factor) had and rewrapped it. I don't think I changed the formula itself. Actually it can even be simplified further:

```js
const currentY = height / 2 * (1 - currentSampleGain * SCALE_FACTOR);
```

Because `MAX_GAIN` is always 1. Right?

Reply:
Gotcha. Thanks for the explanation! But currentSampleGain is a value between -1 to 1. So I assume it will be: 

```
const currentY = height / 2* (1 - (currentSampleGain
        / MAX_GAIN) * SCALE_FACTOR);
```

Reply:
Does `MAX_GAIN`change in the code somewhere? I don't think it's possible because it's `const`. Am I missing something?

If you're thinking about keeping `MAX_GAIN` for readability I am okay with that, but still you're practically dividing something with 1. I was pointing out that's redundant.

Reply:
Okay! I will remove the Max_gain

```
 const currentY = height / 2 * (1 - currentSampleGain * SCALE_FACTOR);
```